crates/utils/src/task_manager/mod.rs

```diff
@@ -187,8 +191,9 @@ impl TaskManager {
             let Some((_name, mut task)) = tasks.remove(&v) else {
                 continue;
             };
+            let handles = task.handle.drain(..);
             task.notifier.notify_waiters();
-            for handle in task.handle.drain(..) {
+            for handle in handles {
```

Comment:
The `is_finished()` function will check the handles in `task.handle`. So if a waiter is notified, assert the tasks `is_finished()` but the handles has not been drained, it may be an assertion fail.

There's a panic log about this. I'm guessing that's what caused the problem 
[panic13.log](https://github.com/user-attachments/files/17553645/panic13.log)